kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -78,10 +80,64 @@ def _get_or_create_uuid() -> str:
         return ""
 
 
+def _get_or_create_project_id(pyproject_path: Path) -> str | None:
+    """
+    Reads a project id from a configuration file or generates and saves a new one if not present.
+    Returns None if configuration file does not exist or does not relate to Kedro.
+    """
+    if pyproject_path.exists():
+        with open(pyproject_path, "r+") as file:
+            pyproject_data = toml.load(file)
+
+            # Check if pyproject related to kedro
+            try:
+                _ = pyproject_data["tool"]["kedro"]
+                try:
+                    project_id = pyproject_data["tool"]["kedro_telemetry"]["project_id"]
+                except KeyError:
+                    project_id = uuid.uuid4().hex
+                    toml_string = (
+                        f'\n[tool.kedro_telemetry]\nproject_id = "{project_id}"\n'
+                    )
+                    file.write(toml_string)
+                return project_id
+            except KeyError:
+                logging.debug(
+                    f"Failed to retrieve project id or save project id: {str(pyproject_path)} does not relate to Kedro"
```

Comment:
Maybe we can make this a bit more specific by mentioning that the file doesn't contain a `tool.kedro` section?